pytensor/tensor/var.py

```diff
@@ -1,1079 +1,11 @@
-import copy
-import traceback as tb
 import warnings
-from collections.abc import Iterable
-from numbers import Number
-from typing import Optional, TypeVar
 
-import numpy as np
 
-from pytensor import tensor as at
-from pytensor.configdefaults import config
-from pytensor.graph.basic import Constant, OptionalApplyType, Variable
-from pytensor.graph.utils import MetaType
-from pytensor.scalar import ComplexError, IntegerDivisionError
-from pytensor.tensor import _get_vector_length
-from pytensor.tensor.exceptions import AdvancedIndexingError
-from pytensor.tensor.type import TensorType
-from pytensor.tensor.type_other import NoneConst
-from pytensor.tensor.utils import hash_from_ndarray
+from pytensor.tensor.variable import *  # noqa
```

Comment:
I was thinking of writing a custom `MetaPathFinder`, `Loader` method to support the import of `var` but then realized we can simply keep `var.py` file with a deprecation warning.
I'm not sure why the mypy is failing, need your help to fix that.

Reply:
Keeping var.py with redirection + warning sounds fine. 

For mypy you may need to add the new file and remove the old one from the "list of files known to fail"

Reply:
https://github.com/pymc-devs/pytensor/blob/main/scripts/mypy-failing.txt

Reply:
Thank you, but why it's failing for `pytensor/tensor/nlinalg.py` ? do I also need to add this file to mypy-failing list?

Reply:
@jessegrabowski can you take a look at the nlinalg mypy failure?

Reply:
It comes from [this line](https://github.com/pymc-devs/pytensor/blob/288a3f3481e7dfa6162e1b60a3909a8f67b29337/pytensor/tensor/nlinalg.py#L302), dating back to 2017. It wasn't failing before?

Reply:
Thank you @jessegrabowski I made this change and it's working fine now